src/components/CommentAuthor.jsx

```diff
@@ -32,8 +32,11 @@ class CommentAuthor extends React.Component {
         this._ismounted = false;
     }
 
-    componentDidUpdate() {
-        this.fetchUserNames();
+    componentDidUpdate(prevProps) {
+        if (prevProps.accessToken !== this.props.accessToken
+            || prevProps.createdBy !== this.props.createdBy) {
```

Comment:
Are you missing here `prevProps.updatedBy !== this.props.updatedBy`?

Reply:
This is not enough, you'd need to check if accessToken is present as well otherwise it would result in 401 and might cause multiple calls to coam.

Reply:
This component does not take updatedBy as a prop.